### ISSUE TITLE
[False-Positive] New value type not in annotated types + preg_replace()

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1112]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False Positive                          |
| Plugin         | Php Inspections (EA Ultimate), 2.0.9 |
| Language level | PHP 7.2                        |

Current behaviour:
---

Please see the following example code

```php
<?php

declare(strict_types=1);

function fooBar1(string $message): void
{
    $message = preg_replace('/\.$/', 'Foo.', $message) ?? '';
}

function fooBar2(string $message): void
{
    $message = preg_replace('/\.$/', 'Foo.', $message);
}

function fooBar3(string $message): string
{
    return preg_replace('/\.$/', 'Foo.', $message);
}
```

For function `fooBar1()` I get this reported: `New value type (array) is not in annotated types.`.

The other 2 functions don't raise a warning regarding that problem (well, besides `$message` being unused in `fooBar2()`). I find that a bit strange - I mean, yes, theoretically `preg_replace()` can return arrays, but not in this case, as the given parameter `$subject` is a string and not an array. So in this case, it can only return a string (or `null` in case of an error). No idea, where the bug here exactly is, because `fooBar2()` and `fooBar3()` are working, and I would have expected they give a warning as well.

Suggested behaviour:
---

Report that `preg_replace()` can return `null` in case of an error and check properly, if the passed `$subject` is a string or an array and adjust warnings accordingly.

**Comment by kalessil** [comment]
Fixed!
